File: 00-Hello-World/py/syntax.py
#!/usr/bin/env python3
"docstring for module."

def subtract(x: int, y: int) -> int:
    "docstring for function."
    # one line comment
    return x - y
    """multiline comment explaining
    that type hints using annotations are not
    enforced but can be used by third party tools
    such as type checkers, IDEs, linters, etc.
    """


if __name__ == "__main__":
    z = subtract(48, 6)
    print("result: {}".format(z))
    txt = f"result of substraction : {z}" # python >= 3.6
    print(txt)

File: 00-Hello-World/rs/syntax.rs
//! docstring for module
//!
//! Compile it with :  rustc syntax.rs
//! Run it : ./syntax

/// docstring for function
fn substract(x: i64, y: i64) -> i64 {
    // one line comment
    x - y
    /* multiline comment explaining
     * that for returning a value
     * an expression as last line is enough
     * (note the absence of ;)
     */
}

fn main() {
    let z = substract(48, 6);
    println!("result of substract : {}", z);
    let txt = format!("result of substract : {z}", z = z);
    println!("{}", txt);
}
